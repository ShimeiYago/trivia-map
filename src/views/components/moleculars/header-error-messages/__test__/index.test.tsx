import { shallow, ShallowWrapper } from 'enzyme';
import { HeaderErrorMessages, Props } from '..';

let wrapper: ShallowWrapper<Props, unknown, HeaderErrorMessages>;

const props: Props = {
  errorTitle: 'Inputted values are invalid.',
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    wrapper = shallow(<HeaderErrorMessages {...props} />);
  });

  it('basic', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('with errorMessages', () => {
    wrapper.setProps({ errorMessages: ['error1'] });
    expect(wrapper).toMatchSnapshot();
  });
});
