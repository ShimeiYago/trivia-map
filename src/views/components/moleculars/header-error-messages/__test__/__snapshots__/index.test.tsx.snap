// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `
<ForwardRef(Alert)
  severity="error"
  sx={
    Object {
      "color": "#f44336",
      "fontWeight": "bold",
    }
  }
>
  Inputted values are invalid.
</ForwardRef(Alert)>
`;

exports[`Shallow Snapshot Tests with errorMessages 1`] = `
<ForwardRef(Alert)
  severity="error"
  sx={
    Object {
      "borderColor": "#f44336",
      "borderStyle": "solid",
      "borderWidth": "2px",
    }
  }
>
  <ForwardRef(AlertTitle)
    sx={
      Object {
        "color": "#f44336",
        "fontWeight": "bold",
      }
    }
  >
    Inputted values are invalid.
  </ForwardRef(AlertTitle)>
  <ul
    className="error-ul"
  >
    <li
      key="error-0"
    >
      error1
    </li>
  </ul>
</ForwardRef(Alert)>
`;
